components/hab/src/cli.rs

```diff
@@ -568,6 +568,20 @@ pub fn get() -> App<'static, 'static> {
                 (@arg SCAFFOLDING: --scaffolding -s +takes_value
                     "Specify explicit Scaffolding for your app (ex: node, ruby)")
             )
+            (@subcommand render =>
+                (about: "Renders plan config files")
+                (aliases: &["r", "re", "ren", "rend", "rende"])
+                (@arg TEMPLATE_PATH: +required +takes_value "Path to config to render")
+                (@arg DEFAULT_TOML: -d --("default-toml") +takes_value "Path to default.toml, defaults to ./default.toml")
+                (@arg USER_TOML: -u --("user-toml") +takes_value "Path to user.toml, defaults to none")
+                (@arg MOCK_DATA: -m --("mock-data") +takes_value "Path to json file with mock data for template, defaults to none")
+                (@arg PRINT: -p --("print") "Prints config to STDOUT")
+                (@arg RENDER_DIR: -r --("render-dir") +takes_value "Path to render templates to, defaults to ./results/")
+                // --no-render doesn't work...
+                (@arg NO_WRITE_FILE: -n --("no-render") --("no-render-dir") "Don't write anything to disk, ignores --render-dir")
```

Comment:
ok, so there's two hard problems in computer science: off by one errors, cache invalidation, and naming things...

My thinking was, by default you probably want to write the config to the filesystem so you can run automated tests against them, and if there's sensitive data then defaulting to _not_ printing the config file to `STDOUT` is the safer option.  Habitat tends to be fairly chatty in general, so "verbose" is the default option, but I probably want to be able to turn of habitat messages and just write the file to disk or `STDOUT`.

So the different combinations might be something like:

* `` (none of these flags) -- standard usage, write config to filesystem, print comfort messages to STDOUT.  I would think for most general debugging and CI use cases this would be the default and allow automated testing on the rendered files.
* `--print` -- print comfort messages, print config to STDOUT, write config to file system.  Good if I want to eyeball the config without opening it in an editor.
* `--print --no-render` -- print comfort messages and config file to STDOUT, don't write anything to filesystem.  Might be used if we're testing in our habitat directory and don't want to remove anything before committing changes.  (I am the _WORST_ about committing test files to repos... my general workflow is: commit changes, push changes, realize I pushed yet another `test.rb` or `foo.rb` that I didn't mean to... `git rm`, commit and force push...)
* `--quiet` -- only write to filesystem.  Might be used in particularly chatty CI environments to cut down on unnecessary output.
* `--quiet --print` -- if we want to print the config file to STDOUT and write it to the filesystem.
* `--quiet --print --no-render` -- only print the config file to STDOUT, don't print comfort messages, don't write to filesystem.  Might be used in other bash (or other) scripting where we capture STDOUT and perform some other operations on the data.  This is probably an edge case but easier than writing to file then loading the contents of that file to memory in another process.  Probably mostly scripted environments.

The names are... kinda bad...  maybe `--no-write-fs` instead of `--no-render`...?

Reply:
I don't have very strong feelings here, so I'll say for the first pass, do the absolutely simplest thing that will meet the requirements. Experience of use from yourself as well as other users will help direct us towards enhancements.

Reply:
Ok, so unless there are any strong feelings, I removed the `--no-render-dir` option so it's just `--no-render`, but otherwise I'm going to leave it as-is for the first pass.